numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
OK, we've now said a lot of things, and I lost track a bit of what the current problem is and/or what the solution would be.

My understanding is still that as written, the tempdir would be removed even if an exception were raised. Is that in dispute? If so, why?

Sorry for pushing the issue, its just that I'm not sure I can fix the problem if I don't understand it, because this does involve both a file that may need to be closed as well as a tempdir that needs to be removed, so with that level of complexity I'm still not sure how to proceed without a better understanding of the issue involved.